### PR TITLE
PRX Auth

### PR DESCRIPTION
**Pull request by cavis**
Port authorization and login components/services from publish.

### CODE REVIEW

File: src/demo/styles.css [hunk 1]
```diff
@@ -49,6 +49,9 @@ ul, li {
   padding-left: 0;
   font-family: Arial, Helvetica, sans-serif;
 }
+pre {
```
Comment (sandikbarr) [inline src/demo/styles.css]: It feels like a bad idea to globally override a block level element to display inline. I don't see this used anywhere? Can you just remove it? If not, it should be scoped with the component it's used in.

File: src/lib/src/auth/auth.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import { ComponentFixture, TestBed, fakeAsync, tick, async } from '@angular/core/testing';
```
Comment (sandikbarr) [inline src/lib/src/auth/auth.component.spec.ts]: fakeAsync and tick are unused imports (but they are a reason I haven't brought over the testing stuff from publish yet, doesn't support these)

File: src/lib/src/auth/login.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import { ComponentFixture, TestBed, fakeAsync, tick, async } from '@angular/core/testing';
```
Comment (sandikbarr) [inline src/lib/src/auth/login.component.spec.ts]: fakeAsync and tick are unused imports

File: src/demo/app/auth/auth-demo.component.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+import { AuthService } from 'ngx-prx-styleguide';
+
+@Component({
+  selector: 'auth-demo',
+  styles: [`
+    .login { width: 400px; }
+    .login b { color: #f00; }
+  `],
+  template: `
+    <h1>AuthModule</h1>
```
Comment (sandikbarr) [inline src/demo/app/auth/auth-demo.component.ts]: 💯 🏅 excellent documentation

File: src/demo/styles.css (hunk 1 above)
Comment (cavis) [inline src/demo/styles.css]: Ah, yeah, i was using `pre` until i saw you using `code`.  Will remove!

**Pull request merged**